### PR TITLE
[Website revamp][11179 - 11180 - 11183] Implemented Quotes component, Header and Navbar

### PR DESCRIPTION
**Pull request by bntnam**
What I've done:
+ Added new component - Quotes for both Desktop/Mobile versions.
+ Updated Hero component (Image Cover) for both Desktop/Mobile versions.
+ Updated Navbar for Desktop version.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Dataflow | Flink | Samza | Spark | Twister2
--- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/) | ---
Java | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Java11/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Twister2/lastCompletedBuild/)
Python | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python38/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/) | ---
XLang | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Direct/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Dataflow/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/) | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website | Whitespace | Typescript
--- | --- | --- | --- | --- | --- | ---
Non-portable | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocker_Cron/lastCompletedBuild/) <br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocs_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Whitespace_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Typescript_Cron/lastCompletedBuild/)
Portable | --- | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | --- | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by bntnam** [comment]
Review: @TheNeuralBit

**Comment by bntnam** [comment]
Hmm. I wonder why the Java check is failed. 🤔

**Comment by TheNeuralBit** [comment]
Run Java PreCommit

File: website/www/site/static/js/hero/hero-mobile.json
Comment (TheNeuralBit) [inline website/www/site/static/js/hero/hero-mobile.json:1]: Is there any way we can get a license header on these? They do seem like creative works

File: website/www/site/layouts/partials/head.html
Comment (TheNeuralBit) [inline website/www/site/layouts/partials/head.html:48]: I think we should be consistent about either serving our own dependencies, or pulling them from CDNs. I'd prefer the latter if possible

File: website/www/site/i18n/home/hero/en.yaml
Comment (TheNeuralBit) [inline website/www/site/i18n/home/hero/en.yaml:20]: Is this actually a move? The files seem unrelated

File: website/www/site/layouts/partials/header.html
Comment (TheNeuralBit) [inline website/www/site/layouts/partials/header.html:109]: This renders strangely for me (on Desktop Chrome):
image

File: website/www/site/layouts/index.html
Comment (TheNeuralBit) [inline website/www/site/layouts/index.html:154]: Is it necessary to have separate quotes-mobile and quotes-mobile-title classes for this part? Couldn't this be handled by CSS?

File: website/www/site/static/js/hero/lottie-light.min.js
Comment (TheNeuralBit) [inline website/www/site/static/js/hero/lottie-light.min.js:1]: This seems to be MIT licensed (based on https://github.com/airbnb/lottie-web), but it doesn't include the license and copyright as it should (for example look at our [bootstrap](https://github.com/apache/beam/blob/master/website/www/site/static/js/bootstrap.min.js). It looks like you just pulled these from https://cdnjs.com/libraries/lottie-web as recommended in airbnb.io/lottie, so maybe its not our problem? I don't know.

This PR is just merging into the website-revamp branch, so we don't need to figure it out right away, but we should figure it out before merging to master.

CC: @kennknowles wdyt?

**Review by TheNeuralBit — COMMENTED** [review]
Thanks @bntnam! a few questions

File: website/www/site/layouts/index.html
Comment (bntnam) [reply website/www/site/layouts/index.html:157]: I tried and I figured out that the external library was using this id to render an element on browser so we should leave it as default. I can write a comment here if you want.

File: website/www/site/layouts/index.html
Comment (bntnam) [reply website/www/site/layouts/index.html:163]: The same answer as above.

File: website/www/site/layouts/partials/hooks/body-end.html
Comment (bntnam) [reply website/www/site/layouts/partials/hooks/body-end.html:17]: Yeah. As you said one of the reason was not to delay the loading of visible content. In this case, there is another reason. For example, `quotes-slider.js` script needs to wrap an element with the id `my-keen-slide` and work on it, then render a new element with such sliding feature. Thus, with this case, the script should be put at the end to wait for the element `my-keen-slider` appears first so that it'll be able to get that element.

File: website/www/site/layouts/partials/header.html
Comment (bntnam) [reply website/www/site/layouts/partials/header.html:109]: Ohh, could you please run it on private mode to make sure that there is no cache? I tested it on Chrome, Firefox, Safari and all fine.

File: website/www/site/static/js/hero/hero-mobile.json
Comment (bntnam) [reply website/www/site/static/js/hero/hero-mobile.json:1]: Unfortunately, we cant :(. If you know another way, please let me know.

Ref: https://stackoverflow.com/questions/244777/can-comments-be-used-in-json#:~:text=JSON%20does%20not%20support%20comments,%2C%20fewer%20mistakes%2C%20more%20comments.

File: website/www/site/static/js/hero/lottie-light.min.js
Comment (bntnam) [reply website/www/site/static/js/hero/lottie-light.min.js:1]: Yeah, the original `lottie-web` on CDN is really heavy, that's why I pulled the light version from their builds. And I followed their instruction here [1].

[1] https://github.com/airbnb/lottie-web#html

File: website/www/site/layouts/index.html
Comment (bntnam) [reply website/www/site/layouts/index.html:154]: Thanks for suggestion!

File: website/www/site/layouts/partials/head.html
Comment (bntnam) [reply website/www/site/layouts/partials/head.html:48]: Done!

File: website/www/site/i18n/home/hero/en.yaml
Comment (bntnam) [reply website/www/site/i18n/home/hero/en.yaml:20]: This file contains texts of Hero component (the Image Cover on header).

File: website/www/site/layouts/index.html
Comment (TheNeuralBit) [reply website/www/site/layouts/index.html]: It looks like "my-keen-slider" and "dots" are both specified in quotes-slider.js which we control, couldn't we change it there? Or are they also specified in the keen slider library?

File: website/www/site/static/js/hero/hero-mobile.json
Comment (TheNeuralBit) [reply website/www/site/static/js/hero/hero-mobile.json:1]: Makes sense. I guess the other option would be to inline this in hero-mobile.js, but then I imagine that makes it harder to work with in the future.

Could you add a TODO (maybe in the build.gradle exclude list that mentions these json files) to figure out if its ok to omit license headers.

File: website/www/site/static/js/hero/hero-desktop.js
Comment (TheNeuralBit) [inline website/www/site/static/js/hero/hero-desktop.js:18]: Does this work if you just specify a relative url, `hero-desktop.json`? As it is it doesn't work on the staged preview since it's in a sub-directoty: http://apache-beam-website-pull-requests.storage.googleapis.com/13439. It tries to access http://apache-beam-website-pull-requests.storage.googleapis.com//js/hero/hero-desktop.json

Alternatively we might inline the JSON with the `animationData` param, but as noted before that has its own drawback.

This wouldn't be a problem in production of course, but it would be nice to keep the preview working.

File: website/www/site/layouts/partials/header.html
Comment (TheNeuralBit) [reply website/www/site/layouts/partials/header.html:109]: Weird I cannot make it go away by avoiding caching. Is it possible it only happens on the staged preview? Could you try looking at it through http://apache-beam-website-pull-requests.storage.googleapis.com/13439

File: website/www/site/static/js/hero/lottie-light.min.js
Comment (TheNeuralBit) [reply website/www/site/static/js/hero/lottie-light.min.js:1]: None of the versions on there actually have a license header.
Can you add a TODO to figure out if we need to do something about this before merging to master?

File: website/www/site/assets/scss/_quotes.scss
Comment (TheNeuralBit) [inline website/www/site/assets/scss/_quotes.scss:154]: nit: I think this should be its own file

**Review by TheNeuralBit — COMMENTED** [review]
Sorry just a couple more requests

File: website/www/site/layouts/index.html
Comment (bntnam) [reply website/www/site/layouts/index.html]: Yeah, they're also specified in the keen slider library. I just moved the CDN to our own file in terms of consistency.

File: website/www/site/static/js/hero/hero-mobile.json
Comment (bntnam) [reply website/www/site/static/js/hero/hero-mobile.json:1]: Ok, a comment added!

File: website/www/site/i18n/home/hero/en.yaml
Comment (bntnam) [reply website/www/site/i18n/home/hero/en.yaml:20]: Oh, I got ur point. It's weird. The `remove` & `add` actions are in separated commits. Idk why git did that behind the scene. Do you have any exp for this case? 😮

File: website/www/site/layouts/partials/header.html
Comment (bntnam) [reply website/www/site/layouts/partials/header.html:109]: I can't access the link

Screen Shot 2020-12-07 at 10 12 22

File: website/www/site/assets/scss/_quotes.scss
Comment (bntnam) [reply website/www/site/assets/scss/_quotes.scss:154]: Done!

File: website/www/site/layouts/partials/header.html
Comment (bntnam) [reply website/www/site/layouts/partials/header.html:109]: Anyway, I guess that I know the problem which occurs on staging. 🤔

File: website/www/site/static/js/hero/lottie-light.min.js
Comment (bntnam) [reply website/www/site/static/js/hero/lottie-light.min.js:1]: Sure!

File: website/www/site/static/js/hero/hero-desktop.js
Comment (bntnam) [reply website/www/site/static/js/hero/hero-desktop.js:18]: Hmm, thanks for your suggestion about `animationData`. I agree that maybe that has its own drawback. However, this approach solves 2 questions (license headers of json files & staging deployment). Then, I'd go for it. 😄

File: website/www/site/layouts/partials/header.html
Comment (TheNeuralBit) [reply website/www/site/layouts/partials/header.html:109]: Ah sorry I think this is the link you need: http://apache-beam-website-pull-requests.storage.googleapis.com/13439/index.html
You can always find it by clicking details on the "Website Stage GCS" jenkins job: 
image

That doesn't need to block this PR though

File: website/www/site/i18n/home/hero/en.yaml
Comment (TheNeuralBit) [reply website/www/site/i18n/home/hero/en.yaml:20]: Interesting. If they're in separate commits, maybe github is just doing this incorrectly?

File: website/www/site/static/js/hero/hero-desktop.js
Comment (TheNeuralBit) [reply website/www/site/static/js/hero/hero-desktop.js]: LG, thanks!

**Review by TheNeuralBit — APPROVED** [review]
LGTM, thank you!

**Pull request merged**
